packages/terra-graphs-docs/src/terra-dev-site/test/graphs/LineGraph/Reflow/Y2Panning.test.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react';
+import LineGraph from '@cerner/terra-graphs/lib/components/Line/LineGraph';
+import Button from 'terra-button/lib/Button';
+import utils from '@cerner/carbon-graphs/lib/js/helpers/utils';
+import '@cerner/terra-graphs-docs/lib/Css/ExampleGraphContainer.module.scss';
+import getConfigLineTimeseriesPanning from '@cerner/terra-graphs-docs/lib/example-datasets/graphConfigObjects/Line/lineTimeseriesPanningY2';
+import data from '@cerner/terra-graphs-docs/lib/example-datasets/dataObjects/Line/y2AxisData';
+
+let graphConfig = utils.deepClone(getConfigLineTimeseriesPanning('#linePanningExample'));
+const dataset = utils.deepClone(data);
+const shift = {
+  initial: 0,
+  factor: 3,
+};
+
+const LinePanningExample = () => {
+  const [panLeftClicked, setPanLeftClicked] = useState(false);
+  const [panRightClicked, setPanRightClicked] = useState(false);
+
+  // Pan left Effect
+  React.useLayoutEffect(() => {
+    if (!panLeftClicked || panRightClicked) {
+      return;
+    }
+
+    const newGraphConfig = utils.deepClone(graphConfig);
+    const hour = shift.initial - shift.factor;
+    shift.initial = hour;
+
+    newGraphConfig.axis.x.lowerLimit = new Date(2016, 0, 1, hour).toISOString();
+    newGraphConfig.axis.x.upperLimit = new Date(2016, 0, 2, hour).toISOString();
+    graphConfig = utils.deepClone(newGraphConfig);
+
+    setPanLeftClicked(false);
+  }, [panLeftClicked, panRightClicked]);
+
+  // Pan right Effect
+  React.useLayoutEffect(() => {
+    if (panLeftClicked || !panRightClicked) {
+      return;
+    }
+
+    const newGraphConfig = utils.deepClone(graphConfig);
+    const hour = shift.initial + shift.factor;
+    shift.initial = hour;
+
+    newGraphConfig.axis.x.lowerLimit = new Date(2016, 0, 1, hour).toISOString();
+    newGraphConfig.axis.x.upperLimit = new Date(2016, 0, 2, hour).toISOString();
+    graphConfig = utils.deepClone(newGraphConfig);
+
+    setPanRightClicked(false);
```

Comment:
While they are functional, these examples with effects that are changing externally defined objects aren't representative of how we'd expect consumers to implement these components. If we moved the config/dataset/shift values themselves into component state, I think we'd get a lot closer to a representative example.

Scenarios like this where one action changes a few different state properties are best served by the useReducer hook: https://reactjs.org/docs/hooks-reference.html#usereducer

With that, the button click handlers could dispatch their appropriate actions, and the reducer you define would manipulate the current state as necessary to cause the graph to update.